pkg/spec/types.go

```diff
@@ -162,6 +162,10 @@ type BuildConfigSpecMod struct {
 
 // ControllerFields are the common fields in every controller Kedge supports
 type ControllerFields struct {
+	// Field to specify the version of application
+	// +optional
+	Appversion string `json:"appversion,omitempty"`
```

Comment:
Hi @piyush1594. You can use `intstr.IntOrString` from `k8s.io/apimachinery/pkg/util/intstr`, check out #455 for inspiration ;-)

Reply:
Hey @kadel 

I tried that function. It is working fine if input is string but if input is int, output is in quotes. And if nothing is in input, output is giving "0".

Reply:
And it gives error in case of float like appversion : 5.6

Reply:
@piyush1594 I think it is throwing "0" because it is converting the zero value of an int (which is 0) to a string.
One solution would be to make the type `*intstr.IntOrString` (because zero value of a pointer is going to be `nil`) and then check the value of `AppVersion` is `nil` or not. If it is `nil`, this means that the user has not set the field.
However, I think that the internal implementation of `instr.IntOrString` is a pointer anyway, so you can directly check if it is `nil` or not, not sure about this though.

Reply:
@kadel for float values like `5.6`, I think we should have something like `FloatOrString`, WDYT?

Reply:
@containscafeine Using pointer it resolved `zero` value problem. Thanks

Reply:
> @kadel for float values like 5.6, I think we should have something like FloatOrString, WDYT?

Hmm, good point. using IntOrString is not going to solve the problem when someone puts something like `5.6` (without quotes) :-(

lets check apimachinery what other types are there besides IntOrString , I bet that we will find something usefull

Reply:
Hmm, more and more I think about this. I would like to stop with this nonsense and just require appversion to be put in quotes as a regular string. 😬

Reply:
We could have the same discussion for every field. This just doesn't make sense. Let's use string.
If the field is required to be a string then it has to be a string and for numbers, quotes are required.

If someone sets label the value to float `5.5` or `5` without quotes,  Kubernetes will also fail (so does Kedge) and that is correct.